README.md

```diff
@@ -2,28 +2,103 @@
-of the platform extend beyond the scope of embedded software development;
-Flipper makes it easy to write applications in any programming language, on any
-platform, that control real hardware.
-
-To purchase a Flipper device, please visit our store
-[here](https://flipper.io/products/flipper-carbon-developer-unit).
+An embedded development platform that can be controlled remotely from any
+programming language. [flipper.io](https://flipper.io)
+
+# Why did we make Flipper?
+
+At the highest level, Flipper is just like any embedded development platform:
+it's a tool for using software to interact with and control electronic
+hardware. All embedded platforms have limited resources, often orders of
+magnitude smaller than those available on a typical desktop computer. These
+limitations impose heavy constraints on the type of applications that you can
+write for these platforms. For example, nearly all embedded programs are
+written in C or C++
+([though Rust is getting there](http://www.rust-embedded.org/)), many lack
+access to resources like heap memory, and simple tasks like communicating with
+other devices can involve a lot of manual protocol implementation. Quite
+frankly, embedded development is really hard, especially for beginners. With
+Flipper, we're trying to make embedded development easier by moving the
+application logic to a more comfortable place - the developer's computer.
+
+# What's special about Flipper?
+
+Flipper has two core components: a lightweight operating system for the
+device, and a library to install on "host machines", such as the developer's
+computer. Together, these form a system that we like to call the Flipper
+Message Runtime, or FMR. FMR allows you to write programs that execute on the
+host machine but directly control the hardware. I like to think of it as
+hardware-as-a-library. This ability to remotely execute functions on the
+device from a host machine has far-reaching implications, and opens up a
+whole new world of possibilities for embedded projects. The most exciting
+feature of Flipper is the ability to control it from any platform (Linux,
+MacOS, Windows, Android, or iOS), using any programming language.
+
+## An example
+
+Flipper has a concept of "packages", which are just compiled code that live
+on the device. The simplest package is the LED, and looks something like this:
+
+```c
+// led.c
+void led_rgb(int r, int g, int b) {
+    // Twiddle some bits, set the LED color
+}
+```
+
+Every package can have language bindings generated for it. This means that
+once we have `led.c`, we can get `led.py` for free:
+
+```py
+# led.py
+from Flipper import led
+led.rgb(0, 255, 0); # Sets the LED to green!
+```
+
+Any language that can execute C under the hood (pretty much all of them!) is
+capable of controlling Flipper like this. So far we're well underway to support
+C/C++, Rust, Python, and Swift, and we plan on adding Objective-C, Java, and
+Javascript in the near future.
+
+# How it works
```

Comment:
This is probably a bit too verbose for the top level README. This would be excellent to put in the Wiki though.

Reply:
On the contrary, I think Flipper is very difficult to explain, so by giving a quick walkthrough of all of the components, it helps to give a better idea of what exactly it does and why you would use it.